assets/scripts/install.ps1

```diff
@@ -1,322 +1,264 @@
+}
 
-    if ($os -eq "windows") {
-        $binsuffix = ".exe"
-        $format = "zip"
+$realTag = Get-RealTag $Tag
+$version = $realTag.TrimStart('v')
+Write-Information "found version ${version} for ${Tag}/${goOS}/${goArch}"
+
+$binarySuffix = ''
+$archiveFormat = 'tar.gz'
+$goOSExtra = ''
+switch ($goOS) {
+    'linux' {
+        $goOSExtra = "-$( Get-LibC )"
+        break
     }
-
-    $github_download = "https://github.com/twpayne/chezmoi/releases/download"
-    New-Item -Type Directory -Path $tempdir | Out-Null
-
-    # download tarball
-    $name="chezmoi_$($version)_$($os)_$($arch)"
-    $tarball="$name.$format"
-    $tarball_url="$($github_download)/$real_tag/$tarball"
-
-    $tmp_tarball = (Join-Path $tempdir $tarball)
-    Fetch-FileFromWeb $tarball_url $tmp_tarball
-
-    # download checksums
-    $checksums = "chezmoi_$($version)_checksums.txt"
-    $checksums_url = "$($github_download)/$($real_tag)/$($checksums)"
-
-    $tmp_checksums = (Join-Path $tempdir $checksums)
-    Fetch-FileFromWeb $checksums_url $tmp_checksums
-
-    # verify checksums
-    verify-hash $tmp_tarball $tmp_checksums
-
-    Push-Location $tempdir
-
-    unpack-file $tarball
-
-    # install the binary
-    if (-not (Test-Path $BinDir)) {
-        New-Item -Type Directory -Path $BinDir | Out-Null
+    'windows' {
+        $binarySuffix = '.exe'
+        $archiveFormat = 'zip'
+        break
     }
+}
+foreach ($variableName in @('binarySuffix', 'archiveFormat', 'goOSExtra')) {
+    Write-DebugVariable $variableName
+}
 
-    $binary = "chezmoi$($binsuffix)";
-    $tmp_binary = (Join-Path $tempdir $binary);
+$archiveFilename = "chezmoi_${version}_${goOS}${goOSExtra}_${goArch}.${archiveFormat}"
+$tempArchivePath = Join-Path -Path $tempDir -ChildPath $archiveFilename
+foreach ($variableName in @('archiveFilename', 'tempArchivePath')) {
+    Write-DebugVariable $variableName
+}
+Invoke-FileDownload "${BaseUrl}/download/${realTag}/${archiveFilename}" $tempArchivePath
 
-    Move-Item -Force -Path $tmp_binary -Destination $BinDir
+$checksums = Get-Checksums $realTag $version
+Confirm-Checksum $tempArchivePath $checksums
 
-    log-info "Installed $($BinDir)/$($binary)"
+Expand-ChezmoiArchive $tempArchivePath
 
-    if ($ExecArgs) {
-        & "$($BinDir)/$($binary)" $ExecArgs
-    }
+$binaryFilename = "chezmoi${binarySuffix}"
+$tempBinaryPath = Join-Path -Path $tempDir -ChildPath $binaryFilename
+foreach ($variableName in @('binaryFilename', 'tempBinaryPath')) {
+    Write-DebugVariable $variableName
 }
+[System.IO.Directory]::CreateDirectory($BinDir) | Out-Null
+$binary = Join-Path -Path $BinDir -ChildPath $binaryFilename
+Write-DebugVariable 'binary'
+Move-Item -Path $tempBinaryPath -Destination $binary -Force
+Write-Information "installed ${binary}"
 
-try {
-    Invoke-Expression ("Install-Chezmoi " + $params)
-} catch {
-    Write-Host "An error occurred while installing: $_"
-} finally {
-    Pop-Location
+Invoke-CleanUp $tempDir

```

Comment:
Is this needed given the

```powershell
trap {
    Invoke-CleanUp $tempDir
    break
}
```

at line 189?

Reply:
The trap will only execute after a terminating error. The "happy path" won't encounter a terminating error, so the trap will never execute, and it won't clean up the temporary archive file.

```console
❯ Get-Content .\test-trap.ps1
trap {
    Write-Host 'exited'
    break
}
Write-Host 'Hello world'

❯ .\test-trap.ps1
Hello world

❯ Get-Content .\test-trap-error.ps1
trap {
    Write-Host 'exited'
    break
}
throw 'Hello error'
Write-Host 'Hello world'

❯ .\test-trap-error.ps1
exited
Exception: C:\Users\Braden\test-trap-error.ps1:5
Line |
   5 |  throw 'Hello error'
     |  ~~~~~~~~~~~~~~~~~~~
     | Hello error
```

Reply:
Thank you for the explanation :)